src/bootstrap/compile.rs

```diff
@@ -60,6 +60,11 @@ impl Step for Std {
         let target = self.target;
         let compiler = self.compiler;
 
+        if builder.config.download_stage1 {
```

Comment:
It's a bit surprising to me to see this here but not in the other steps (e.g., rustc or codegen)...

Reply:
The other steps are never called when `download-stage1` is set (because `Assemble` skips them). This is only called because `Std` is built by default when you run `x.py build`.

I guess that means that `x.py build compiler/rustc` will do unnecessary work when it could just copy the files over, but I don't know why anyone would need that, especially in a first draft.

Reply:
Hmm, now that you mention it, I think I can simplify this by calling `run.default_condition(!config.download_stage1)` for Std.

Reply:
That seems wrong. It shouldn't matter, I imagine, whether std is being requested by say rustdoc build internally or is requested on the command line.

Reply:
My thought was that if you run `x.py build --stage 1`, you want to test `cfg(not(bootstrap))` changes to the standard library. But if you prefer, I can add this check to all the steps so they don't run if `download-stage1` is set (maybe with another `ENABLE_STAGE1` constant in `trait Step` so I don't have to play wack-a-mole).

Reply:
Hm, if we can make that work, I guess it would be good - I'm not quite sure we can pull it off though.

Reply:
Hmm, so this ended up feeling a little constrictive, since only Rustdoc and check targets are enabled. It does work, though.
```
> xpy build
Skipping bootstrap::compile::CodegenBackend because it is not enabled for `download-stage1`
Building rustdoc for stage1 (x86_64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.20s
Build completed successfully in 0:00:01
> xpy build compiler/rustc
Skipping bootstrap::compile::Rustc because it is not enabled for `download-stage1`
> xpy build library/std
Skipping bootstrap::compile::Std because it is not enabled for `download-stage1`
```

Reply:
I think this is fine, but we'll want to polish the language a bit - maybe something like "Not running $step because its artifacts have been downloaded from CI, as download-stage1 is set".

It might also be good to rename to download-rustc or something like it, since we're not really downloading stage1 so much as downloading a rustc that can be used for any stage, really.

Reply:
Good idea, done.